### DISCUSSION TITLE
Which code formatter or style guide should we all use?

### DISCUSSION THREAD

**Discussion by joncodo** [apache/devlake#98] [General]
@All I see we are doing different formatting of files. I think we should all be on the same VS code extension with the same config. I am using prettier with default configs. Does anyone have other suggestions?

https://github.com/prettier/prettier-vscode

Its super easy to use with VSCode and we can enforce it but you can also use it with NPM and VIM

**Comment by basicthinker** [comment]
@hackwaly what is the current standard for EE?

**Reply by hackwaly** [reply]
We use eslint as code formatter for EE. Some developers use editors other than vscode, eg. sublime and idea. It is harder to  setup eslint for those editors. Maybe this is the cause of the mess code style of the codebase.

**Comment by hackwaly** [comment]
> @ALL I see we are doing different formatting of files. I think we should all be on the same VS code extension with the same config. I am using prettier with default configs. Does anyone have other suggestions?
> 
> https://github.com/prettier/prettier-vscode
> 
> Its super easy to use with VSCode and we can enforce it but you can also use it with NPM and VIM

I recommend eslint-config-standard or eslint-config-airbnb
